akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala

```diff
@@ -239,12 +251,44 @@ class PersistentBehavior[Command, Event, State](
     copy(tagger = tagger)
 
   private def copy(
-    persistenceIdFromActorName: String ⇒ String                       = persistenceIdFromActorName,
-    initialState:               State                                 = initialState,
-    commandHandler:             CommandHandler[Command, Event, State] = commandHandler,
-    eventHandler:               (State, Event) ⇒ State                = eventHandler,
-    recoveryCompleted:          (ActorContext[Command], State) ⇒ Unit = recoveryCompleted,
-    tagger:                     Event ⇒ Set[String]                   = tagger,
-    snapshotOn:                 (State, Event, Long) ⇒ Boolean        = snapshotOn): PersistentBehavior[Command, Event, State] =
-    new PersistentBehavior(persistenceIdFromActorName, initialState, commandHandler, eventHandler, recoveryCompleted, tagger, snapshotOn)
+    initialState:      State                                 = initialState,
+    commandHandler:    CommandHandler[Command, Event, State] = commandHandler,
+    eventHandler:      (State, Event) ⇒ State                = eventHandler,
+    recoveryCompleted: (ActorContext[Command], State) ⇒ Unit = recoveryCompleted,
+    tagger:            Event ⇒ Set[String]                   = tagger,
+    snapshotWhen:      (State, Event, Long) ⇒ Boolean        = shouldSnapshot,
+    journalPluginId:   String                                = journalPluginId,
+    snapshotPluginId:  String                                = snapshotPluginId): PersistentBehavior[Command, Event, State] =
+    new PersistentBehavior[Command, Event, State](
+      _persistenceId = _persistenceId,
+      initialState = initialState,
+      commandHandler = commandHandler,
+      eventHandler = eventHandler,
+      recoveryCompleted = recoveryCompleted,
+      tagger = tagger,
+      journalPluginId = journalPluginId,
+      snapshotPluginId = snapshotPluginId,
+      shouldSnapshot = shouldSnapshot)
+
+}
+
+// this is only convenience as it's a pain to create the abstract one directly in the immutable factory method
+final class InitialPersistentBehavior[Command, Event, State](
+  override val context:  ActorContext[Any],
+  override val timers:   TimerScheduler[Any],
+  _persistenceId:        String,
+  val initialState:      State,
+  val commandHandler:    PersistentBehaviors.CommandHandler[Command, Event, State],
+  val eventHandler:      (State, Event) ⇒ State,
+  val recoveryCompleted: (ActorContext[Command], State) ⇒ Unit,
+  val tagger:            Event ⇒ Set[String],
+  val journalPluginId:   String,
+  val snapshotPluginId:  String,
+  val shouldSnapshot:    (State, Event, Long) ⇒ Boolean
+) extends EventsourcedRequestingRecoveryPermit[Command, Event, State](context) {
+
+  override val persistenceId: String =
+    if (_persistenceId.trim.isEmpty) context.self.path.name
```

Comment:
rather not, I think empty persistenceId should be disallowed

Reply:
Ok, will throw eagerly if that's done then